lsp/server/src/utils/orgUtils.ts

```diff
@@ -116,18 +76,26 @@ export class OrgUtils {
         }
     }
 
-    // Updates the auth state async
+    // Update the auth state async
     private static async checkAuthStatus(): Promise<AuthStatus> {
         if (this.authStatus !== AuthStatus.UNKNOWN) {
             return this.authStatus;
         }
+        try {
+            this.orgName = await this.getDefaultOrg();
+        } catch (e) {
+            this.orgName = undefined;
+        }
         const connection = await this.getConnection();
         if (connection === undefined) {
             //It is possible that orgName exists and connection expires
-            this.orgName = '';
+            this.orgName = undefined;
             this.authStatus = AuthStatus.UNAUTHORIZED;
         } else {
             this.authStatus = AuthStatus.AUTHORIZED;
+            if (this.orgName === undefined) {
+                throw new Error('Authorized Org does not have org name!');
+            }
```

Comment:
Is it possible to have `undefined` org name when the auth status is `AUTHORIZED`?

Reply:
Not possible. This is the safeguard which correlates to the OrgName change

Reply:
If it is not possible why bother with having a safeguard?

Reply:
In theory, it is not possible. The safeguard code is put there in case some bug comes in and we know what the problem is. I will take it out.

Reply:
The code is taken out